From eb20a8202e6194c1e52a1ba9e77daf419bcdd0d9 Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Sat, 14 Mar 2020 21:06:33 +1300
Subject: futures: tests/eventual.rs : Disable w/o executor + thread-pool

This test requires both executor and thread-pool enabled to function.

But as the code is all heavily shared and intertwined, it seems the
most sensible thing to do is just file-level disabling.

Spurious build failures no longer occur with:

  cargo hack --each-feature \
    --skip bilock \
    --skip cfg-target-has-atomic \
    --skip read-initializer \
    test --test eventual

  cargo hack --feature-powerset \
    --skip bilock \
    --skip cfg-target-has-atomic \
    --skip read-initializer \
    test --test eventual

  cargo hack --features "unstable" \
    --each-feature \
    --skip "unstable" \
    test --test eventual

  cargo hack --features "unstable" \
    --feature-powerset \
    --skip unstable \
    test --test eventual
---
 tests/eventual.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/eventual.rs b/tests/eventual.rs
index bff000d..6835588 100644
--- a/tests/eventual.rs
+++ b/tests/eventual.rs
@@ -1,3 +1,4 @@
+#![cfg(all(feature = "executor", feature = "thread-pool"))]
 use futures::channel::oneshot;
 use futures::executor::ThreadPool;
 use futures::future::{self, ok, Future, FutureExt, TryFutureExt};
-- 
2.25.0

